test/reset.spec.js

```diff
@@ -0,0 +1,369 @@
+  href: "https://example.com/login",
+};
+
+// Mock API response
+const mockResponse = {
+  data: {
+    tokens: {
+      access: { value: createAccessToken() },
+      id: { value: createIdToken() },
+      refresh: { value: createRefreshToken() },
+    },
+    nonce: "nonce-value",
+    redirectTo: "/dashboard",
+  },
+};
+
+describe("sendResetLink", () => {
+  beforeEach(() => {
+    Userfront.init(tenantId);
+  });
+
+  it(`error should respond with whatever the server sends`, async () => {
+    // Mock the API response
+    const mockResponse = {
+      response: {
+        data: {
+          error: "Bad Request",
+          message: `That's a silly link request.`,
+          statusCode: 400,
+        },
+      },
+    };
+    axios.post.mockImplementationOnce(() => Promise.reject(mockResponse));
+    expect(sendResetLink("email@example.com")).rejects.toEqual(
+      new Error(mockResponse.response.data.message)
+    );
+  });
+
+  it(`should respond with whatever the server sends when using custom baseUrl`, async () => {
+    Userfront.init(tenantId, {
+      baseUrl: customBaseUrl,
+    });
+
+    // Mock the API response
+    const mockResponse = {
+      response: {
+        data: {
+          error: "Bad Request",
+          message: `That's a silly link request.`,
+          statusCode: 400,
+        },
+      },
+    };
+    axios.post.mockImplementationOnce(() => Promise.reject(mockResponse));
+
+    const email = "email@example.com";
+    expect(sendResetLink(email)).rejects.toEqual(
+      new Error(mockResponse.response.data.message)
+    );
+
+    // Should have sent the proper API request
+    expect(axios.post).toHaveBeenCalledWith(`${customBaseUrl}auth/reset/link`, {
+      email,
+      tenantId,
+    });
+  });
+});
+
+describe("resetPassword", () => {
+  it("should be an alias for updatePassword()", () => {
+    expect(resetPassword).toEqual(updatePassword);
+  });
+});
+
+describe("updatePassword", () => {
+  beforeEach(() => {
+    Userfront.init(tenantId);
+  });
+
+  afterEach(() => {
+    window.location.assign.mockClear();
+  });
+
+  describe("No method set (method is inferred)", () => {
+    it("should call updatePasswordWithLink() if JWT access token is not present", async () => {
+      // Mock the API response
+      axios.put.mockImplementationOnce(() => Promise.resolve(mockResponse));
+
+      const options = { token: "token", uuid: "uuid", password: "password" };
+
+      // Call updatePassword
+      await updatePassword(options);
+
+      // Should have sent the proper API request
+      expect(axios.put).toHaveBeenCalledWith(
+        `https://api.userfront.com/v0/auth/reset`,
+        {
+          tenantId,
+          ...options,
+        }
+      );
+
+      // Should have redirected the page
+      expect(window.location.assign).toHaveBeenCalledWith(
+        mockResponse.data.redirectTo
+      );
+    });
+
+    it("should call updatePasswordWithJwt() if JWT access token is present", async () => {
```

Comment:
Similar comment to the last test, this test is good but the description could be clearer. In this case it calls `updatePasswordWithJwt()` if a JWT access token is present _and_ link credentials are not present.

To make the target behavior really well-defined, if you want there could be a third test to assert that if both credential types are present, `updatePasswordWithLink()` is called.

Reply:
I added an additional test that covers the case where a JWT access token is present, yet the `token` and `uuid` are still taken from the URL.